crates/sui-config/src/builder.rs

```diff
@@ -96,6 +99,11 @@ impl<R> ConfigBuilder<R> {
         self
     }
 
+    pub fn with_objects(mut self, objects: Vec<Object>) -> Self {
```

Comment:
Rather than `Vec` in particular, could this be `IntoIterator` to match the signature of `extend`?